app/src/Root/index.js

```diff
@@ -249,10 +249,42 @@ const client = new ApolloClient({
 
 const moduleLoadTime = Date.now();
 
+const MarketTable = ({
+  CategoricalMarketTable,
+  ScalarMarketTable,
+  ...props
+}) => {
+  const { markets } = props;
+  if (markets && markets.length > 0) {
+    if (markets[0].type === "CATEGORICAL") {
+      return <CategoricalMarketTable {...props} />;
+    } else if (markets[0].type === "SCALAR") {
+      return <ScalarMarketTable {...props} />;
+    } else {
+      throw Error("Unknown market type");
+    }
+  }
+};
+
+const Sidebar = ({ CategoricalSidebar, ScalarSidebar, ...props }) => {
+  const { markets } = props;
+  if (markets && markets.length > 0) {
+    if (markets[0].type === "CATEGORICAL") {
+      return <CategoricalSidebar {...props} />;
+    } else if (markets[0].type === "SCALAR") {
+      return <ScalarSidebar {...props} />;
+    } else {
+      throw Error("Unknown market type");
+    }
+  }
```

Comment:
When no markets are available this component would error instead of displaying nothing. Return `null` or maybe render an error.